### DISCUSSION TITLE
How to support Super key?

### DISCUSSION THREAD

**Discussion by plmi** [bakkeby/st-flexipatch#105] [Q&A] [answered]
Hi,
I want to use emacs in terminal mode (`-nw`) but I can't use the `Alt` key as `Meta` key because it conflicts with dwm. I've tried to bind `Meta` to `Super_L` but `st` doesn't seem to support that. I've found `FIXKEYBOARDINPUT_PATCH` which seems to be exactly what I'm looking for but I don't know how to set it up correctly. How can I make `st` support `Super_L` with this patch?

**Comment by bakkeby** [comment, marked as answer]
> I've tried to bind Meta to Super_L but st doesn't seem to support that.

If I am understanding this correctly you are changing the emacs configuration to set Super_L as the Meta modifier, but st does not seem to forward those keys to the terminal application?

I'm not sure whether the fix keyboard input patch can be used to address your issue or not so can't really comment on that.

I found this page which outlines a bunch of meta key issues including xterm:
https://www.emacswiki.org/emacs/MetaKeyProblems

> I want to use emacs in terminal mode (-nw) but I can't use the Alt key as Meta key because it conflicts with dwm.

Alt, Ctrl and Shift are very common modifiers used for keyboard shortcuts for applications. 

I don't know why dwm use Alt as the primary modifier by default - I can only imagine that the original author had a keyboard where that button had the windows logo and as such that key was a no-go.

In general I recommend to use the Super key as the primary modifier for a window manager as it creates less conflicts. Alternatively you can enable mod3 or mod5 using `xmodmap`.

**Reply by plmi** [reply]
Ok so for anybody else having that problem here is what I did. I have a keyboard that is programmable with VIA. I've swapped the mapping of my `Super_L` and `ALT` keys. I've additionally updated my dwm `config.h` and use `Mod4Mask` (`Super_L`) as my `MODKEY` now. So now I can use `ALT` for emacs. As I've switched the two mappings of `ALT` and `Super_L` with VIA I can still control dwm with the same physical key on my keyboard as before so I don't have to adapt to new keys.

Thanks for the quick help!
